src/app/project/documents/documents-tab.component.ts

```diff
@@ -183,6 +183,11 @@ export class DocumentsTabComponent implements OnInit, OnDestroy {
     this.documentsGroupedBySection = this.documentsGroupedBySection.filter((document) => {
       return Array.isArray(document);
     })
```

Comment:
You can just tack on the `map` method like you would with chaining jquery methods. You can save a bit of code here

```suggestion
    }).map(section => 
      section.sort((a: any, b: any) => new Date(b.dateAdded || b.datePosted).getTime() - new Date(a.dateAdded || a.datePosted).getTime())
    );
```

Reply:
I think it might be better to leave this code organized by purpose, if you don't mind.

// Filter out sections with no files in them.
    this.documentsGroupedBySection = this.documentsGroupedBySection.filter((document) => {
      return Array.isArray(document);
    })

    // Sort the grouped documents.
    this.documentsGroupedBySection = this.documentsGroupedBySection.map(section => 
      section.sort((a: Document, b: Document) => new Date(b.dateAdded || b.datePosted).getTime() - new Date(a.dateAdded || a.datePosted).getTime())
    );

Reply:
Not a biggie in this case, but if it's just for readability then you can still leave comments to delineate the different steps/purposes.